### ISSUE TITLE
properties could change err.status

### ISSUE THREAD

**Issue by zensh** [jshttp/http-errors#16]
the test case will failed in `koa`:
```js
describe('when props include status', function(){
    it('should be ignored', function(done){
      var ctx = context();

      try {
        ctx.throw(400, 'msg', {
          prop: true,
          status: -1
        });
      } catch (err) {
        assert('msg' == err.message);
        assert(400 == err.status);
        assert(true === err.expose);
        assert(true === err.prop);
        done();
      }
    })
  })
```

Is it intentional?

**Comment by dougwilson** [comment]
Sorry, it wasn't clear from the description, but it looks like it was a regression, basically. It'll be fixed :)

**Comment by dougwilson** [comment]
Fix published as 1.3.1 :) Sorry for the regression; added a test for this as well :)
